core/src/main/java/org/infinispan/marshall/core/ClassIdentifiers.java

```diff
@@ -0,0 +1,71 @@
+package org.infinispan.marshall.core;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.container.entries.InternalCacheValue;
+
+public class ClassIdentifiers implements InternalClassIds {
+   private final static ClassIdentifiers INSTANCE = new ClassIdentifiers();
```

Comment:
Btw, the biggest danger here is if the collection keeps track of user classes. If they're JDK or Infinispan classes it should be fine, but to be on the safe side, I'd avoid any static strong references to classes.